components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -217,18 +289,40 @@ func newRTKMovementSensor(
 	}
 
 	// baud rate
-	g.wbaud = newConf.NtripBaud
-	if g.wbaud == 38400 {
+	if newConf.NtripBaud == 0 {
+		newConf.NtripBaud = 38400
 		g.logger.Info("ntrip_baud using default baud rate 38400")
 	}
+	g.wbaud = newConf.NtripBaud
 
-	if g.writepath != "" {
-		g.logger.Info("ntrip_path will use same path for writing RCTM messages to gps")
-		g.writepath = newConf.NtripPath
+	if g.inputProtocol == serialStr {
```

Comment:
linter may complain about this not being a switch statement, I also find them more readable

Reply:
This being lines 298 - 304 and 308 to 325